dali/pipeline/executor/executor.h

```diff
@@ -142,6 +147,16 @@ class DLL_PUBLIC Executor : public ExecutorBase, public QueuePolicy {
 
   DISABLE_COPY_MOVE_ASSIGN(Executor);
 
+  /**
+   * Returns the current checkpoint.
+  */
+  DLL_PUBLIC Checkpoint& GetCurrentCheckpoint();
+
+  /**
+   * Restores states of operators.
+  */
+  DLL_PUBLIC void RestoreStateFromCheckpoint(const Checkpoint &cpt);
```

Comment:
We should consider guarding against using the Restore when the executor was started/is running. It should only be allowed before any operator was executed, right?